### ISSUE TITLE
Master: Scan not working correctly

### ISSUE THREAD

**Issue by RandolfCarter** [owncloud/music#374]
### Steps to reproduce
1. Download latest master.zip
2. Disable old music app
2. unzip master.zip to music folder in owncloud/apps
3. Enable music app

### Expected behaviour
When navigating to Music app, all music is shown (or newly scanned).

### Actual behaviour
Only very few pieces of music are shown. In my case, this was two files from within a Document folder (not in the configured music collection path), and when re-selecting the collection folder, the one file residing directly in the collection folder, but no file in any subfolder. There was no indication that the scan wasn't finished yet. I also tried the command line, with the same results.

In contrast, with the 0.3.5 app, all music is found.

### Server configuration
**Operating system**: Ubuntu 14.04.2 LTS
**Web server:** Apache 2.4.12
**Database:** MariaDB Ver 15.1 Distrib 10.0.16
**PHP version:** 5.5.22
**ownCloud version:** 8.0.2
**Updated from an older ownCloud or fresh install:** updated
**Are you using external storage, if yes which one:** Music is located on local storage!
**Are you using encryption:** no
**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Firefox 37.0.2
**Operating system:** Linux 15.04

#### Web server error log
-

#### ownCloud log (data/owncloud.log)
Some Rescan triggered / Rescan finished messages
And:
```
"Error while running background job: An exception occurred while executing
'UPDATE `oc_music_albums` SET `cover_file_id` = NULL
WHERE `cover_file_id` IS NOT NULL AND `cover_file_id` IN
(SELECT `cover_file_id` FROM `oc_music_albums` LEFT JOIN `oc_filecache`
ON `cover_file_id`=`fileid` WHERE `fileid` IS NULL);'
:\n\nSQLSTATE[HY000]: General error: 1093 Table 'oc_music_albums' is specified twice, both as a target for 'UPDATE' and as a separate source for data"
```

#### Browser log
-

**Comment by RandolfCarter** [comment]
The SQL error points at [this commit](https://github.com/owncloud/music/commit/9771cc0954163fd7216fe43a26a2381d39be2420), and replacing the first changed statement (the UPDATE) with the old one seems to fix the problem...

**Comment by CommanderAlchemy** [comment]
Also got this after updating owncloud, any news on this?

**Comment by MorrisJobke** [comment]
@butonic Maybe you have a clever idea how to fix this for mysql?

**Comment by CommanderAlchemy** [comment]
Seems to be a limitation in mysql update: Currently, you cannot update a table and select from the same table in a subquery.

http://dev.mysql.com/doc/refman/5.6/en/update.html

So it seems it should be split into two queries sep.

**Comment by MorrisJobke** [comment]
Fixed with #372

**Comment by RandolfCarter** [comment]
Looks good, system is scanning happily again :)
